File: app/src/main/java/com/pleon/buyt/ui/activity/SettingsActivity.java
package com.pleon.buyt.ui.activity;

import android.content.SharedPreferences;
import android.graphics.drawable.Animatable;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import com.google.android.material.bottomappbar.BottomAppBar;
import com.pleon.buyt.R;

import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.PreferenceManager;
import butterknife.BindView;
import butterknife.ButterKnife;

import static com.pleon.buyt.ui.activity.MainActivity.DEFAULT_THEME;
import static com.pleon.buyt.ui.activity.MainActivity.KEY_PREF_THEME;

public class SettingsActivity extends AppCompatActivity {

    @BindView(R.id.bottom_bar) BottomAppBar bottomAppBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme();
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        ButterKnife.bind(this);

        setSupportActionBar(bottomAppBar);
    }

    private void setTheme() {
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
        String theme = preferences.getString(KEY_PREF_THEME, DEFAULT_THEME);
        setTheme(DEFAULT_THEME.equals(theme) ? R.style.AppTheme : R.style.LightTheme);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_bottom_settings, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_reset:
                ((Animatable) item.getIcon()).start();
                break;
            case android.R.id.home:
                finish();
        }
        return true;
    }
}

File: app/src/main/kotlin/com/pleon/buyt/ui/activity/SettingsActivity.kt
package com.pleon.buyt.ui.activity

import android.view.Menu
import android.view.MenuItem
import android.widget.TextView
import androidx.preference.PreferenceManager
import com.pleon.buyt.R
import com.pleon.buyt.ui.fragment.PREF_NEWBIE
import com.pleon.buyt.util.AnimationUtil.animateIcon

class SettingsActivity : BaseActivity() {

    private lateinit var resetMenuItemView: TextView

    override fun layout() = R.layout.activity_settings

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_bottom_settings, menu)

        // Setting up "Reset" action because it has custom layout
        val resetMenuItem = menu.findItem(R.id.action_reset)
        resetMenuItemView = resetMenuItem.actionView as TextView
        resetMenuItemView.setOnClickListener { onOptionsItemSelected(resetMenuItem) }

        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_reset -> {
                animateIcon(resetMenuItemView.compoundDrawablesRelative[2])
                resetPreferences()
            }
            android.R.id.home -> finish()
        }
        return true
    }

    private fun resetPreferences() {
        prefs.edit().clear().apply()
        PreferenceManager.setDefaultValues(this, R.xml.preferences, true)
        prefs.edit().putBoolean(PREF_NEWBIE, false).apply() // We don't want tutorial again!
    }
}
